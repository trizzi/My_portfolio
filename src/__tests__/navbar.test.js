import renderer from 'react-test-renderer';
import { Link } from 'react-scroll';

it('changes the class when menu is open', () => {
	const componetLinks = renderer.create(
		<Link
			to='section1'
			toggle={() => {}}
			closeMenu={() => {}}>
			Home
		</Link>,
		<Link
			to='section2'
			toggle={() => {}}
			closeMenu={() => {}}>
			About
		</Link>,
		<Link
			to='section3'
			toggle={() => {}}
			closeMenu={() => {}}>
			Projects
		</Link>,
		<Link
			to='section4'
			toggle={() => {}}
			closeMenu={() => {}}>
			Services
		</Link>,
	);

	let tree = componetLinks.toJSON();
	expect(tree).toMatchSnapshot();

	// manually trigger the callback
	renderer.act(() => {
		tree.props.toggle();
	});
	// re-rendering
	tree = componetLinks.toJSON();
	expect(tree).toMatchSnapshot();

	// manually trigger the callback
	renderer.act(() => {
		tree.props.closeMenu();
	});
	// re-rendering
	tree = componetLinks.toJSON();
	expect(tree).toMatchSnapshot();
});
