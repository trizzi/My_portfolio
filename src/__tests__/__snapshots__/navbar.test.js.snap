// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the class when menu is open 1`] = `
<a
  closeMenu={[Function]}
  onClick={[Function]}
  style={Object {}}
  toggle={[Function]}
>
  Home
</a>
`;

exports[`changes the class when menu is open 2`] = `
<a
  closeMenu={[Function]}
  onClick={[Function]}
  style={Object {}}
  toggle={[Function]}
>
  Home
</a>
`;

exports[`changes the class when menu is open 3`] = `
<a
  closeMenu={[Function]}
  onClick={[Function]}
  style={Object {}}
  toggle={[Function]}
>
  Home
</a>
`;
